lib/core/connection/utils.js

```diff
@@ -31,21 +42,43 @@ function noSnappyWarning() {
   );
 }
 
+const PKG_VERSION = Symbol('kPkgVersion');
+
 // Facilitate loading Snappy optionally
 function retrieveSnappy() {
-  let snappy = require_optional('snappy');
+  const snappy = require_optional('snappy');
   if (!snappy) {
-    snappy = {
+    return {
       compress: noSnappyWarning,
       uncompress: noSnappyWarning,
       compressSync: noSnappyWarning,
       uncompressSync: noSnappyWarning
     };
   }
+
+  const snappyPkg = require_optional('snappy/package.json') || { version: '0.0.0' };
+  const version = parsePackageVersion(snappyPkg);
+  snappy[PKG_VERSION] = version;
+  if (version.major >= 7) {
+    const compressOriginal = snappy.compress;
+    const uncompressOriginal = snappy.uncompress;
+    snappy.compress = (data, callback) => {
```

Comment:
Slightly different approach taken, just wrap the functions here, the actual usage sight needs no changes now.